modules/ROOT/pages/dataweave-types.adoc

```diff
@@ -732,8 +732,7 @@ var y = {e: "e"}
   (y)
 }
 ----
-
-It is important to note that the expression between the _parentheses_ (such as `(x)` where `x` is the variable shown in the header) should return an array of objects. All of objects in that array get merged together. They are also merged with the contained object. So the output looks like this:
+Notice the expression between the _parentheses_ (such as `(x)` where `x` is the variable shown in the header) should return a single object or an array of objects. In this example, `(x)` returns and array of objects, and `(y)` returns an object with just one key/value. All of the key/values from each object in that array get flattened "in place" by the object constructor curly braces into the outer object. The object constructor curly braces also operate on the `(y)` expression to flatten the single key/value "in place" from that object into the outer object. So the output looks like this:
```

Comment:
```suggestion
Notice the expression between the _parentheses_ (such as `(x)` where `x` is the variable shown in the header) returns a single object or an array of objects. In this example, `(x)` returns an array of objects, and `(y)` returns an object with just one key-value pair. The object constructor curly braces flattens all of the key-value pairs from each object "in place," into the outer object. The object constructor curly braces also operate on the `(y)` expression to flatten the single key-value pair from that object into the outer object. The output looks like this:
```

Reply:
Notice the expression between the _parentheses_ (such as `(x)` where `x` is the variable shown in the header) returns a single object or an array of objects.

Reply:
Wondering if something like this would work:

The _evaluation parentheses_ around the variables (`(x)` and `(y)`) in the body of the script above enable the object constructor curly braces (`{ }`) that surrounds the body of the script to extract the key-value pairs from the values of `x` and `y` and convert those values into a collection of key-value pairs within an object. Without the parentheses (for example, if you use `x` instead of `(x)`), the script returns an error `Invalid input '}', expected Namespace or Attribute&lt;'@('(Name:Value)+')'&gt; `.

The object constructor curly braces require a data construct that contains key-value pairs, specifically an object, such as `{ "a":"one", "b":"two"}`, or an array of objects, such as `[{"a":"one"},{"b":"two"}]` so that it can produce a valid object. Because the first element within the object constructor curly braces is the key-value pair `a: "a"`, the object constructor curly braces leaves it as the first element in the output object, without any modifications. However, the object constructor curly braces extracts the child elements of the `(x)` and `(y)` values from the containing array or object and lists them in the output object.